src/yunohost/permission.py

```diff
@@ -140,12 +140,14 @@ def user_permission_update(operation_logger, permission, add=None, remove=None,
     # If we end up with something like allowed groups is ["all_users", "volunteers"]
     # we shall warn the users that they should probably choose between one or the other,
     # because the current situation is probably not what they expect / is temporary ?
-
-    if len(new_allowed_groups) > 1:
-        if "all_users" in new_allowed_groups:
+    
+    if "all_users" in new_allowed_groups:
+        if len(new_allowed_groups) > 1 and "visitors" not in new_allowed_groups or len(new_allowed_groups) > 2:
             logger.warning(m18n.n("permission_currently_allowed_for_all_users"))
-        if "visitors" in new_allowed_groups:
-            logger.warning(m18n.n("permission_currently_allowed_for_visitors"))
+    
+    # If visitors are allowed, but not all users, it can break some applications, so we prohibit it.
+    if "visitors" in new_allowed_groups and "all_users" not in new_allowed_groups:
+        raise YunohostError('permission_allowed_for_visitors_but_not_for_all_users')
```

Comment:
Here I don't know what to do in terms of design / UX ... Shouldn't we implictly add `all_users` and display a warning that we did add it automatically ?

That way running `yunohost permission update foo.main --add visitors` will work whatever the current state we're in

Reply:
I don't know if it's really clear for the user (my current implementation isn't). What to do in case of removing `all_users`? Should we remove `visitors` too?

Reply:
Well let's go for this : 
- when adding `visitors`, it implicitly adds `all_users` if needed
- when removing `all_users`, it raises an exception if `visitors` is allowed, advising the user to remove it first.